test/async_cache_test.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:async/async.dart';
+import 'package:test/test.dart';
+
+void main() {
+  AsyncCache<String> cache;
+  DateTime time;
+
+  setUp(() {
+    // Create a cache that is fresh for an hour starting at 1/1/17 @ 12:30PM.
+    cache = new AsyncCache<String>(const Duration(hours: 1), now: () => time);
+    time = new DateTime(2017, 1, 1, 12, 30);
+  });
+
+  test('should fetch via a callback when no cache exists', () async {
+    var callback = expectAsync0/*<Future<String>>*/(() async => 'Expensive');
+    expect(await cache.fetch(callback), 'Expensive');
+  });
+
+  test('should not fetch via callback when a cache exists', () async {
+    Future<String> doNotCall() async => 'Should not have been invoked...';
```

Comment:
Make this inline, and use `expectAsync` to assert that it's actually not called (e.g. `expectAsync0(() {}, count: 0)`). Same for callbacks below that aren't intended to be called.